File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0]*n
        i, j = 0, n-1
        for k in range(n-1, -1, -1):
            left, right = nums[i]*nums[i], nums[j]*nums[j]
            if left < right:
                res[k] = right
                j -= 1
            else:
                res[k] = left
                i += 1
        return res

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int i {};
        int j { static_cast<int>(nums.size()-1)};
        vector<int> res(nums.size());
        int k { j };
        while (i<=j) {
            if (nums[i]*nums[i] < nums[j]*nums[j]) {
                res[k--] = nums[j]*nums[j--];
            }
            else {
                res[k--] = nums[i]*nums[i++];
            }
        }
        return res;
    }
};